src/mscorlib/src/System/Threading/Semaphore.cs

```diff
@@ -132,7 +132,7 @@ private static OpenExistingResult OpenExistingWorker(string name, out Semaphore
             throw new PlatformNotSupportedException(SR.PlatformNotSupported_NamedSynchronizationPrimitives);
 #else
             if (name == null)
-                throw new ArgumentNullException(nameof(name), SR.ArgumentNull_WithParamName);
+                throw new ArgumentNullException(nameof(name), SR.Format(SR.ArgumentNull_WithParamName, nameof(name)));
```

Comment:
Should we rather delete the `ArgumentNull_WithParamName` message, and use the standard ArgumentNullException formatting here?